addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,882 @@
+                // special case when the last welcome message is a forward to an operator
+                // we need to save the welcome messages before continuing the script
+                // indeed, if there are no operator available, the script will continue
+                // with steps that are NOT included in the welcome messages
+                // (hence why we need to have those welcome messages posted BEFORE that)
+                this._chatbotCheckWelcomeMessages();
+            }
+
+            // we are done posting welcome messages, let's start the actual script
+            this._chatbotProcessStep();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Restart the script and then trigger the "next step" (which will be the first of the script
+     * in this case).
+     *
+     * @private
+     */
+    _onChatbotRestartScript: async function (ev) {
+        this._chatWindow.$('.o_composer_text_field').removeClass('d-none');
+        this._chatWindow.$('.o_livechat_chatbot_end').addClass('d-none');
+
+        if (this.nextStepTimeout) {
+            clearTimeout(this.nextStepTimeout);
+        }
+
+        if (this.welcomeMessageTimeout) {
+            clearTimeout(this.welcomeMessageTimeout);
+        }
+
+        const postedMessage = await session.rpc('/chatbot/restart', {
+            channel_uuid: this._livechat.getUUID(),
+            chatbot_script_id: this._chatbot.chatbot_script_id,
+        });
+
+        if (postedMessage) {
+            this._chatbotAddMessage(postedMessage);
+        }
+
+        this._chatbotCurrentStep = null;
+        this._chatbotSetIsTyping();
+        this.nextStepTimeout = setTimeout(
+            this._chatbotTriggerNextStep.bind(this), this._chatbotMessageDelay);
+    },
+
+    _onChatbotInputKeyDown: function () {
+        if (this._chatbotCurrentStep &&
+            this._chatbotCurrentStep.chatbot_step_type === 'free_input_multi') {
+            this._debouncedChatbotAwaitUserInput();
+        }
+    },
+
+    /**
+     * We need to save the selected answer right away to be safe in case we redirect to another page
+     * Indeed, on the other page, we need to know that an answer was selected to avoid showing
+     * the options again when we re-open the conversation with the bot.
+     *
+     * @param {MouseEvent} ev
+     * @private
+     */
+    _onChatbotOptionClicked: async function (ev) {
+        ev.stopPropagation();
+
+        const $target = $(ev.currentTarget);
+        const stepId = $target.closest('ul').data('chatbotStepId');
+        const selectedAnswer = $target.data('chatbotStepAnswerId');
+
+        const redirectLink = $target.data('chatbotStepRedirectLink');
+        if (redirectLink) {
+            this._isChatbotRedirecting = true;
```

Comment:
``this._isChatbotRedirecting = redirectLink !== null`` (or equivalent that works in JS, should be able to do bool(), saleté de language) ? That way the variable is defined with true / false, instead of true / undefined.

Reply:
> or equivalent that works in JS, should be able to do bool(), saleté de language)

`!!foo` is the conventional way to say `bool(foo)`.